src/renderer/app.jsx

```diff
@@ -43,8 +43,26 @@ document.body.appendChild(appTarget);
 
 GUI.setAppElement(appTarget);
 
-const ScratchDesktopHOC = function (WrappedComponent) {
-    class ScratchDesktopComponent extends React.Component {
+const ScratchDesktopOuterHOC = function (WrappedComponent) {
+    const ScratchDesktopOuterComponent = function (props) {
+        const shouldShowTelemetryModal = (typeof ipcRenderer.sendSync('getTelemetryDidOptIn') !== 'boolean');
+
+        return (<WrappedComponent
+            canEditTitle
+            canModifyCloudData={false}
+            isScratchDesktop
```

Comment:
These 3 props seem like they belong in the inner HOC since one could see the role of this outer HOC as to pass props into `AppStateHOC`. If that's the case then I think it would be clearer if we only had things necessary for AppStateHOC in this outer HOC, maybe accompanied by a more specific name like `ScratchDesktopAppStateHOC`?